nemo_curator/classifiers/fineweb_edu.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+os.environ["RAPIDS_NO_INITIALIZE"] = "1"
+os.environ["DASK_DATAFRAME__QUERY_PLANNING"] = "False"
+import torch
+from crossfit import op
+from crossfit.backend.torch.hf.model import HFModel
+from transformers import AutoConfig, AutoModelForSequenceClassification
+
+from nemo_curator.classifiers.base import (
+    DistributedDataClassifier,
+    _get_suggest_memory_for_classifier,
+    _run_classifier_helper,
+)
+from nemo_curator.datasets import DocumentDataset
+
+FINEWEB_EDU_IDENTIFIER = "HuggingFaceTB/fineweb-edu-classifier"
+
+
+class FinewebEduModel(HFModel):
+    def __init__(self, path_or_name, max_mem_gb, autocast=False):
+        self.path_or_name = path_or_name
+        self.autocast = autocast
+        if max_mem_gb is None:
+            max_mem_gb = _get_suggest_memory_for_classifier()
+        super().__init__(path_or_name=path_or_name, max_mem_gb=max_mem_gb)
+
+    def load_model(self, device="cuda"):
+        model = AutoModelForSequenceClassification.from_pretrained(self.path_or_name)
+        model = model.to(device)
+        model = self.configure_forward(model, self.autocast)
+        return model
+
+    @staticmethod
+    def configure_forward(model, autocast=True):
+        original_forward = model.forward
+
+        def custom_forward(*args, **kwargs):
+            if autocast:
+                with torch.autocast(device_type="cuda"):
+                    output = original_forward(*args, **kwargs)
+            return output.logits.squeeze(-1).float()
+
+        model.forward = custom_forward
+        return model
+
+    def load_config(self):
+        return AutoConfig.from_pretrained(self.path_or_name)
+
+
+class FineWebEduClassifier(DistributedDataClassifier):
+    def __init__(
+        self,
+        filter_by=None,
+        batch_size=256,
+        text_field: str = "text",
+        pred_column="fineweb-edu-score",
```

Comment:
The way I had `AegisClassifier` was that `pred_column` was the final processed output and `raw_pred_column` was an intermediate result. Is `"fineweb-edu-score"` meant to be the final processed output and `"fineweb-edu-score-int"` a more raw form? If so, I think it would be good to follow the pattern in Aegis.

Reply:
So it is a decimal score which they round.  I dont want it to say raw, was trying to follow the classifier. Dont think `raw` tells the right story.  So i went with `score` and `score-int`, can change it to say `fineweb-edu-int-score` if thats better. 

```
result = {
    "text": text,
    "score": score,
    "int_score": int(round(max(0, min(score, 5)))),
}
```

See below: https://huggingface.co/HuggingFaceFW/fineweb-edu-classifier

Reply:
Have added docstring, hopefully that makes it easier to understand.

Reply:
Oh gotcha, so it just outputs 2 scores. Makes sense.